modules/authentication/proc-enabling-authentication-with-microsoft-azure.adoc

```diff
@@ -0,0 +1,130 @@
+. You link:https://docs.redhat.com/en/documentation/red_hat_developer_hub/{product-version}/html-single/administration_guide_for_red_hat_developer_hub/index#assembly-add-custom-app-file-openshift_admin-rhdh[added a custom {product-short} application configuration].
+
+.Procedure
+. To allow {product-short} to authenticate with Microsoft Azure, create an OAuth Application in Microsoft Azure.
+
+.. Go to link:https://portal.azure.com/#view/Microsoft_AAD_RegisteredApps/ApplicationsListBlade[*Azure Portal > App registrations*], and create an *App Registration* for {product-short}.
+
+.. On your *App registration* overview page, add a new *Web platform configuration*, with the configuration:
+
+*Redirect URI*:: Enter the backend authentication URI set in {product-short}: `pass:c,a,q[https://_<APP_FQDN>_/api/auth/microsoft/handler/frame]`
+*Front-channel logout URL*:: Leave blank.
+*Implicit grant and hybrid flows*:: Leave all checkboxes cleared.
+
+.. On the *API permissions* tab, click *Add Permission*, then add the following *Delegated permission* for the
+*Microsoft Graph API*:
++
+* `email`
+* `offline_access`
+* `openid`
+* `profile`
+* `User.Read`
+* Optional custom scopes of the Microsoft Graph API that you define both here and in the {product-short} configuration (`app-config-rhdh.yaml`).
++
+[NOTE]
+====
+Your company might require you to grant admin consent for these permissions.
+Even if your company does not require admin consent, you might do so as it means users do not need to individually consent the first time they access backstage.
+To grant admin consent, a directory admin must go to the link:https://learn.microsoft.com/en-us/azure/active-directory/manage-apps/user-admin-consent-overview[admin consent] page and click *Grant admin consent for COMPANY NAME*.
+====
+
+
+.. Go to the *Certificates & Secrets* page, then the *Client secrets* tab, and create a new client secret.
+
+.. Save for the next step:
+- **Directory (tenant) ID**
+- **Application (client) ID**
+- **Application (client) secret**
+
+. To add your Microsoft Azure credentials to {product-short}, add the following key/value pairs to your {product-short} secrets, such as `secrets-rhdh`:
++
+`AUTH_AZURE_TENANT_ID`:: Enter your saved *Directory (tenant) ID*.
+`AUTH_AZURE_CLIENT_ID`:: Enter your saved *Application (client) ID*.
+`AUTH_AZURE_CLIENT_SECRET`:: Enter your saved *Application (client) secret*.
+
+. Set up the Microsoft Azure authentication provider in your {product-short} custom configuration, such as `app-config-rhdh`:
++
+--
+.`app-config-rhdh.yaml` fragment
+[source,yaml,subs="+quotes,+attributes"]
+----
+auth:
+  environment: production
+  providers:
+    microsoft:
+      production:
+        clientId: ${AUTH_AZURE_CLIENT_ID}
+        clientSecret: ${AUTH_AZURE_CLIENT_SECRET}
+        tenantId: ${AUTH_AZURE_TENANT_ID}
+signInPage: microsoft
+dangerouslyAllowSignInWithoutUserInCatalog: true
```

Comment:
It's probably better to suggest to configure the ingestion and this parameter to false first, then mention the `dangerouslyAllowSignInWithoutUserInCatalog` as a secondary option: I don't think it is intended as a production configuration.